### ISSUE TITLE
The main documentation should be update-to-date

### ISSUE THREAD

**Issue by ngoan1608** [test-fullautomation/RobotFramework_AIO#214]
Hi Thong,

As our discussion, the main documentation (and its relevant files) which is generated by `genmaindoc` within build pipeline should be checked-in back to [robotframework-documentation](https://github.com/test-fullautomation/robotframework-documentation).

Hi @ThPoll ,
I and Thong are concerning about the PR to check-in main documentation, will we do it for all builds or just for `rel/dev` build?

Thank you,
Ngoan

**Comment by test-fullautomation** [comment]
Hi Ngoan,
`rel/dev` builds are enough.
Thank you,
 Thomas

**Comment by huavanthong** [comment]
Hi Thomas,
Cc @HolQue , @ngoan1608 

As my observation, after the tool **genmaindoc.py** in [robotframework-documentation](https://github.com/test-fullautomation/robotframework-documentation) is executed completely, I see that there are some changed files below:
```
Modified Files:
README.md
RobotFrameworkAIO/Components.html
RobotFrameworkAIO/RobotFrameworkAIO_Reference.pdf
RobotFrameworkAIO/components.rst

Untracked files:
book/BundleVersionDate.tex
book/RobotFrameworkAIO_Reference.pdf
```
Could you help me confirm that which files we need to update for automation??

Thank you and best regards,
Thong Hua

**Comment by HolQue** [comment]
Hi Thong,

the book folder contain source files and temp files. This is the development area. Finally (after genmaindoc is ready) all results are copied to folder RobotFrameworkAIO. Therefore relevant for the build process (and for users) are the following files:

```
robotframework-documentation/RobotFrameworkAIO/Components.html
robotframework-documentation/RobotFrameworkAIO/Components.rst
robotframework-documentation/RobotFrameworkAIO/RobotFrameworkAIO_Reference.pdf
```

These three files have to be part of the AIO distribution. And now it came into my mind: The idea behind the files Components.html/rst is to be able to provide a direct link to the information inside. Therefore at end of build process all three files should be uploaded to the documentation repository (and not the pdf only).

The build process can ignore the following files inside the book folder. They are autogenerated and temporary:

```
robotframework-documentation/book/BundleVersionDate.tex
robotframework-documentation/book/RobotFrameworkAIO_Reference.pdf
```

Also the entire

`robotframework-documentation\book\externaldocs`

folder is a temporary one and can be ignored by the build process.

The README.md is converted out of the README.rst. When I update the rst file, I make a genmaindoc run to get the md file updated also. Therefore they are in sync in the repository. But nevertheless you will get the md file recreated in case you also run genmaindoc. But as per my understanding there is no need to add the README files to the AIO distribution. Therefore I propose the build process to ignore the README files.

**Comment by huavanthong** [comment]
Hi Holger,

Thank you for your confirmation. I will only update the 3 files as your mention:
```
robotframework-documentation/RobotFrameworkAIO/Components.html
robotframework-documentation/RobotFrameworkAIO/Components.rst
robotframework-documentation/RobotFrameworkAIO/RobotFrameworkAIO_Reference.pdf
```

Thank you and best regards,
Thong Hua

**Comment by huavanthong** [comment]
Hi Thomas,
Cc all,

I would like to summary the solution to resolve this ticket.
**Robotframework_AIO**
* Updating documents will be triggered by a tag 'rel' or 'dev'.
* After build process, it will update all changed files in robotframework-documentation and push to branch:
`doc_branch = htv3hc/feat/auto-update-doc`
* Then, it will trigger robotframework-documentation job for creating a pull request at branch `htv3hc/feat/auto-update-doc`
* More details: https://github.com/test-fullautomation/RobotFramework_AIO/pull/220

**robotframework-documentation**
* Receive the event from RobotFramework AIO repository.
* Trigger job to create a pull request with all changed documents. 
* More details: https://github.com/test-fullautomation/robotframework-documentation/pull/51

Please take it for your information.
If you have any concerns, please let me know.

Thank you and best regards,
Thong Hua

**Comment by test-fullautomation** [comment]
Hi Thong,
looks like a good solution to me.
I merge this for 0.10.0.
Thank you,
 Thomas

**Comment by test-fullautomation** [comment]
merged for 0.10.0
